### DISCUSSION TITLE
Client request editor

### DISCUSSION THREAD

**Discussion by StringNick** [ogen-go/ogen#1135] [General]
I tried to research how i can integrate change header or add some logger as server middleware to client library in dev environments, and i researched now it's not possible, so is it possible to add something like oapi-codegen for client generated code, for editing request (for example add custom header, and etc).
Example editor for special route and call:
<img width="767" alt="image" src="https://github.com/ogen-go/ogen/assets/13052752/993e3c3e-a279-4a20-a6e6-52260c0af631">
Example of global request editor for all calls:
```go
client, clientErr := NewClient("", WithRequestEditorFn(someHeaderOrLogEditor))
```
It can help to add logger, to add custom debug headers that u cant specify in spec and etc.
Without this configuration its inpossible to use generator for client in my project, for server i saw solution about abstract on server interface, but for client nothing.
Thank you for attention

**Comment by ghost** [comment]
For now, `ogen` generates a `WithClient` option, you can make a custom [`ht.Client`](https://pkg.go.dev/github.com/ogen-go/ogen/http#Client) implementation wrapping `http.Client`.

**Reply by StringNick** [reply]
yes this solution is global, thank you, but what about local solution in methods, is it possible or its against philosophy

**Reply by ghost** [reply]
I opened a feature request (#1138) for that. I think it is possible as opt-in feature.

**Reply by StringNick** [reply]
ty
